src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/ReferenceList.cs

```diff
@@ -0,0 +1,140 @@
+using System.Collections.Generic;
+using CoreWCF.IdentityModel;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using DictionaryManager = CoreWCF.IdentityModel.DictionaryManager;
+using ISecurityElement = CoreWCF.IdentityModel.ISecurityElement;
+using CoreWCF;
+using XD = CoreWCF.IdentityModel.XD;
+using System;
+
+namespace CoreWCF.Security
+{
+   sealed class ReferenceList : ISecurityElement
+    {
```

Comment:
You should only need the statics ElementName and NamespaceUri. Everything else isn't needed as SignedXml has it's own equivalent of this.

Reply:
I looked at this a bit closer and this is being used with the EncryptedKey class. We don't need to have our own implementation of this. There's an implementation of EncryptedKey in the System.Security.Cryptography.Xml namespace and it looks like it has all the same features as the implementation in this codebase and should be an easy switch out..